### DISCUSSION TITLE
send files to /home/jovyan from VSCode

### DISCUSSION THREAD

**Discussion by tlohde** [microsoft/PlanetaryComputer#215] [Q&A] [answered]
Hello PC team,

My query is related to [this discussion](https://github.com/microsoft/PlanetaryComputer/discussions/7), from a year and a bit ago, only less security focused.

I use, and access PC, primarly through VSCode ([as outlined here](https://planetarycomputer.microsoft.com/docs/overview/ui-vscode/)).

When running a locally stored notebook is it possible to import a locally stored module? I _think_ the answer is no. So, is there a way to easily access the files on `/home/jovyan` from _within_ VSCode, and or easily push local files to the hub without having to drag and drop..? With the goal of developing a module using VSCode, sending it up to the hub, to then run the notebook from within VSCode that is connected to the PC Jupyter server. Will following the instructions outlined on the [jupyterhub-ssh](https://github.com/yuvipanda/jupyterhub-ssh) page work with PC's JupyterHub?

Thank you,
tlohde

**Comment by TomAugspurger** [comment, marked as answer]
> I think the answer is no.

I also think the answer is "no", but I'm not 100% sure about that. My recollection is that the notebooks the Jupyter extension for VS Code edits are local to your VS Code. Even if you connect to a remote Kernel running on JupyterHub, it doesn't necessarily expose the file system associated with that JuptyerHub account. But that's all speculation on my part.

I think that jupyterhub-ssh needs to be installed / enabled on the Hub, so that might not be an option.

I'm not aware of programmatic way to post files to your JupyterHub's home directory.

**Reply by tlohde** [reply]
Hi Tom, 
That's a shame. Oh well.
Thank you for the quick reply.
Cheers,
